test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java

```diff
@@ -558,9 +561,17 @@ public void indexStatementsWithConditions() throws Throwable
     @Test
     public void indexCorrectlyMarkedAsBuildAndRemoved() throws Throwable
     {
-        String selectBuiltIndexesQuery = String.format("SELECT * FROM %s.\"%s\"",
+        String selectBuiltIndexesQuery = String.format("SELECT table_name, index_name FROM %s.\"%s\"",
                                                        SchemaConstants.SYSTEM_KEYSPACE_NAME,
                                                        SystemKeyspace.BUILT_INDEXES);
+
+        // Wait for any background index clearing tasks to complete. Warn: When we used to run tests in parallel there
+        // could also be cross test class talk and have other indices pop up here.
+        Awaitility.await()
+                  .atMost(1, TimeUnit.MINUTES)
+                  .pollDelay(1, TimeUnit.SECONDS)
+                  .untilAsserted(() -> assertRows(execute(selectBuiltIndexesQuery)));
+
         UntypedResultSet rs = execute(selectBuiltIndexesQuery);
```

Comment:
Saving the `initialSize` is part of the fix introduced by CASSANDRA-15565 that we can get rid of. The checks after dropping the int can be simplified to just verifying that the `IndexInfo` table is empty, as suggested [here](https://github.com/adelapena/cassandra/blob/8cf7145b8437f616e8085169cc3829197961031c/test/unit/org/apache/cassandra/index/internal/CassandraIndexTest.java#L567-L584).